drf_rw_serializers/generics.py

```diff
@@ -25,6 +30,42 @@ def get_serializer_class(self):
 
         return self.serializer_class
 
+    def get_serializer_class(self):
+        """
+        Return the class to use for the serializer.
+        Defaults to using `self.serializer_class`.
+        If the request method is GET, it tries to use `self.read_serializer_class`.
+        If the request method is not GET, it tries to use `self.write_serializer_class`.
+        If the specific serializer class for the request method is not set, it falls back to
+        `self.serializer_class`.
+        You may want to override this if you need to provide different
+        serializations depending on the incoming request.
+        (Eg. admins get full serialization, others get basic serialization)
+        """
+        if hasattr(self, "request"):
+            if self.request.method in ["GET"]:
```

Comment:
Do you know if HEAD, OPTIONS or TRACE would get the write serializer? I imagine they should get the read one, but they don't have a body, so I'm not sure if it would really do anything.

Reply:
Yes, currently they would get the write serializer. Should I include those values in the condition?

Reply:
I think so. These should be read.

Reply:
Related change: commit [58e95c3](https://github.com/vintasoftware/drf-rw-serializers/pull/77/commits/58e95c350dc43437fa7dc6a3d485ec982de04e81). Let me know if there are any adjustments to make to it.